cross-media-measurement-api/src/main/proto/wfa/measurement/api/v2alpha/measurement.proto

```diff
@@ -143,7 +144,30 @@ message Measurement {
   // is `SUCCEEDED`.
   //
   // The symmetric encryption key is derived using a key-agreement protocol
-  // between `measurement_public_key` in `metric_definition` and
+  // between `measurement_public_key` in `measurement_spec` and
   // `result_public_key`.
   bytes encrypted_result = 11;
 }
+
+// Specification for a `Measurement`. Immutable.
+message MeasurementSpec {
+  // Encryption public key for the `Measurement` that this `MeasurementSpec` is
+  // associated with. Required.
+  bytes measurement_public_key = 1;
+
+  message EncryptedSketch {
+    // Resource key of the sketch configuration. Required.
+    SketchConfig.Key sketch_config = 1;
+
+    // The resource key of the `CombinedPublicKey` that is used to encrypt the
+    // sketch. Required.
+    CombinedPublicKey.Key combined_public_key = 2;
+  }
+
+  // What data the requisition is for, i.e. what data is being requisitioned.
+  // Required.
+  oneof for {
```

Comment:
The generated code will lead to stuff like `switch (spec.getForCase())`, which feels a little awkward, but I don't have a better name.